wormhole-connect/src/utils/localStorage.ts

```diff
@@ -1,56 +1,71 @@
 import { TransactionLocal } from 'config/types';
 
-// Adds an object to localStorage
-export const addItemToLocalStorage = (
-  id: string, // Key for the item
-  data: object, // Item data
-  prefix: string, // Prefix to use for the item key
-  max?: number, // Max number of items allowed with the same prefix
-) => {
+const LOCAL_STORAGE_KEY = 'wormhole-connect:transactions:inprogress';
+const LOCAL_STORAGE_MAX = 3;
+
+// Retrieves all in-progress transactions from localStorage
+export const getTxsFromLocalStorage = ():
+  | Array<TransactionLocal>
+  | undefined => {
   const ls = window.localStorage;
-  const items: Array<string> = [];
 
-  // Get all current items with the prefix
+  // Find the in-progress transactions list in localStorage
   for (let i = 0; i < ls.length; i++) {
     const itemKey = ls.key(i);
-    if (itemKey?.toLowerCase().startsWith(prefix)) {
-      items.push(itemKey);
+    if (itemKey?.toLowerCase() === LOCAL_STORAGE_KEY) {
+      const item = ls.getItem(itemKey);
+      if (item) {
+        try {
+          return JSON.parse(item);
+        } catch (e: any) {
+          console.log(
+            `Error while parsing localStorage item ${LOCAL_STORAGE_KEY}: ${e}`,
+          );
+          return;
+        }
+      }
     }
   }
+};
 
-  // Remove the oldest items that puts the length at or over the limit
-  if (max && items.length >= max) {
-    for (let i = 0; i < items.length - max + 1; i++) {
-      ls.removeItem(items[i]);
-    }
+// Adds a TransactionLocal object to localStorage
+export const addTxToLocalStorage = (
+  data: TransactionLocal, // Item data
+) => {
+  const ls = window.localStorage;
+  const items: Array<TransactionLocal> | undefined = getTxsFromLocalStorage();
```

Comment:
in general it's preferable to add the return type to the function signature (`getTxsFromLocalStorage`) rather than leaving the function return value untyped and having to annotate the type in the assignment here.

Reply:
Oh the return type is already there in the function definition, I forgot to remove the types from the callers.